pallets/phala/src/lib.rs

```diff
@@ -100,5 +125,48 @@ decl_module! {
 
 			Ok(())
 		}
+
+		#[weight = 0]
+		fn transfer_to_tee(origin, #[compact] amount: BalanceOf<T>) -> dispatch::DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			T::Currency::transfer(&who, &Self::account_id(), amount, AllowDeath)
+				.map_err(|_| dispatch::DispatchError::Other("Can't transfer to tee"))?;
+
+			Self::deposit_event(RawEvent::TransferToTee(who.encode(), amount));
+
+			Ok(())
+		}
+
+		#[weight = 0]
+		//fn transfer_to_chain(origin, dest: T::AccountId, #[compact] amount: BalanceOf<T>) -> dispatch::DispatchResult {
+		fn transfer_to_chain(origin, data: Vec<u8>) -> dispatch::DispatchResult {
+			ensure_signed(origin)?;
+
+			let transfer_data: TransferData<<T as system::Trait>::AccountId, BalanceOf<T>> = Decode::decode(&mut &data[..]).unwrap();
+			let id = transfer_data.id;
+			let sequence = Sequence::get();
+			ensure!(transfer_data.sequence == sequence + 1, "bad sequence");
+
+			T::Currency::transfer(&Self::account_id(), &transfer_data.dest, transfer_data.amount, AllowDeath)
+				.map_err(|_| dispatch::DispatchError::Other("Can't transfer to chain"))?;
+
+			Sequence::set(sequence + 1);
+
+			Self::deposit_event(RawEvent::TransferToChain(id.to_vec(), transfer_data.dest.encode(), transfer_data.amount, sequence + 1));
```

Comment:
Why the second param `transfer_data.dest.encode()` is needed? Can we just pass `transfer_data.dest`?